### ISSUE TITLE
On upgrade from 1.7.3 to 1.8.0-rc1, file protocol_features cannot be found

### ISSUE THREAD

**Issue by matthewdarwin** [EOSIO/eos#7239]
I upgraded my existing installation by replacing the nodeos 1.7.3 binary with 1.8.0-rc1 binary and started it how I normally start it, using the same config file.  It generates the following error:

```
Apr 29 19:55:53 jungle-peer6 nodeos[10978]: APPBASE: Warning: The following configuration items in the config.ini file are redundantly set to
Apr 29 19:55:53 jungle-peer6 nodeos[10978]:          their default value:
Apr 29 19:55:53 jungle-peer6 nodeos[10978]:              read-mode, validation-mode, max-body-size, allowed-connection, connection-cleanup-period,
Apr 29 19:55:53 jungle-peer6 nodeos[10978]:              network-version-match, enable-stale-production, max-transaction-time, max-irreversible-block-age,
Apr 29 19:55:53 jungle-peer6 nodeos[10978]:              txn-reference-block-lag
Apr 29 19:55:53 jungle-peer6 nodeos[10978]:          Explicit values will override future changes to application defaults. Consider commenting out or
Apr 29 19:55:53 jungle-peer6 nodeos[10978]:          removing these items.
Apr 29 19:55:53 jungle-peer6 nodeos[10978]: info  2019-04-29T19:55:53.814 nodeos    http_plugin.cpp:465           plugin_initialize    ] configured http to listen on 127.0.0.1:8888
Apr 29 19:55:53 jungle-peer6 nodeos[10978]: info  2019-04-29T19:55:53.814 nodeos    chain_plugin.cpp:557          plugin_initialize    ] initializing chain plugin
Apr 29 19:55:53 jungle-peer6 nodeos[10978]: warn  2019-04-29T19:55:53.814 nodeos    chain_plugin.cpp:959          plugin_initialize    ] 13 N5boost10filesystem16filesystem_errorE: boost::filesystem::create_directory: No such file or directory: "/home/nodeos/.local/share/eosio/nodeos/config/protocol_features"
Apr 29 19:55:53 jungle-peer6 nodeos[10978]: rethrow boost::filesystem::create_directory: No such file or directory: "/home/nodeos/.local/share/eosio/nodeos/config/protocol_features":
Apr 29 19:55:53 jungle-peer6 nodeos[10978]:     {"what":"boost::filesystem::create_directory: No such file or directory: \"/home/nodeos/.local/share/eosio/nodeos/config/protocol_features\""}
Apr 29 19:55:53 jungle-peer6 nodeos[10978]:     nodeos  chain_plugin.cpp:959 plugin_initialize
Apr 29 19:55:53 jungle-peer6 nodeos[10978]: error 2019-04-29T19:55:53.814 nodeos    main.cpp:120                  main                 ] 13 N5boost10filesystem16filesystem_errorE: boost::filesystem::create_directory: No such file or directory: "/home/nodeos/.local/share/eosio/nodeos/config/protocol_features"
Apr 29 19:55:53 jungle-peer6 nodeos[10978]: rethrow boost::filesystem::create_directory: No such file or directory: "/home/nodeos/.local/share/eosio/nodeos/config/protocol_features":
Apr 29 19:55:53 jungle-peer6 nodeos[10978]:     {"what":"boost::filesystem::create_directory: No such file or directory: \"/home/nodeos/.local/share/eosio/nodeos/config/protocol_features\""}
Apr 29 19:55:53 jungle-peer6 nodeos[10978]:     nodeos  chain_plugin.cpp:959 plugin_initialize
```

The file "protocol_features" is not listed anywhere in the release notes as to what it should contain.

**Comment by arhag** [comment]
Is your `/home/nodeos/.local/share/eosio/nodeos/config/` read-only? Nodeos is supposed to automatically create a folder called `protocol_features` inside the config directory if it doesn't already exist.

**Comment by matthewdarwin** [comment]
The directory doesn't exist actually.  /home/nodeos is writeable by tye nodeos user.

**Comment by matthewdarwin** [comment]
Doing this first, fixes this problem:

```
mkdir -p /home/nodeos/.local/share/eosio/nodeos/config/
chown -R nodeos.nodeos /home/nodeos/.local
```

**Comment by matthewdarwin** [comment]
I re-built all the libraries when changing from 1.7 to 1.8, which presumably results in this problem?

```
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.074 nodeos    http_plugin.cpp:465           plugin_initialize    ] configured http to listen on 127.0.0.1:8888
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.074 nodeos    chain_plugin.cpp:557          plugin_initialize    ] initializing chain plugin
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.074 nodeos    chain_plugin.cpp:409          operator()           ] Support for builtin protocol feature 'PREACTIVATE_FEATURE' (with digest of '0ec7e080177b2c02b278d5088611686b49d739925a92d9bfcacd7fc6b74053bd') is enabled without activation restrictions
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.074 nodeos    chain_plugin.cpp:396          operator()           ] Support for builtin protocol feature 'ONLY_LINK_TO_EXISTING_PERMISSION' (with digest of '1a99a59d87e06e09ec5b028a9cbb7749b4a5ad8819004365d02dc4379a8b7241') is enabled with preactivation required
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.074 nodeos    chain_plugin.cpp:396          operator()           ] Support for builtin protocol feature 'FORWARD_SETCODE' (with digest of '2652f5f96006294109b3dd0bbde63693f55324af452b799ee137a81a905eed25') is enabled with preactivation required
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.074 nodeos    chain_plugin.cpp:396          operator()           ] Support for builtin protocol feature 'REPLACE_DEFERRED' (with digest of 'ef43112c6543b88db2283a2e077278c315ae2c84719a8b25f25cc88565fbea99') is enabled with preactivation required
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.074 nodeos    chain_plugin.cpp:396          operator()           ] Support for builtin protocol feature 'NO_DUPLICATE_DEFERRED_ID' (with digest of '4a90c00d55454dc5b059055ca213579c6ea856967712a56017487886a4d4cc0f') is enabled with preactivation required
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.074 nodeos    chain_plugin.cpp:396          operator()           ] Support for builtin protocol feature 'RAM_RESTRICTIONS' (with digest of '4e7bf348da00a945489b2a681749eb56f5de00b900014e137ddae39f48f69d67') is enabled with preactivation required
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.074 nodeos    chain_plugin.cpp:396          operator()           ] Support for builtin protocol feature 'DISALLOW_EMPTY_PRODUCER_SCHEDULE' (with digest of '68dcaa34c0517d19666e6b33add67351d8c5f69e999ca1e37931bc410a297428') is enabled with preactivation required
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.074 nodeos    chain_plugin.cpp:396          operator()           ] Support for builtin protocol feature 'ONLY_BILL_FIRST_AUTHORIZER' (with digest of '8ba52fe7a3956c5cd3a656a3174b931d3bb2abb45578befc59f283ecd816a405') is enabled with preactivation required
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.075 nodeos    chain_plugin.cpp:396          operator()           ] Support for builtin protocol feature 'RESTRICT_ACTION_TO_SELF' (with digest of 'ad9e3d8f650687709fd68f4b90b41f7d825a365b02c23a636cef88ac2ac00c43') is enabled with preactivation required
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.075 nodeos    chain_plugin.cpp:396          operator()           ] Support for builtin protocol feature 'FIX_LINKAUTH_RESTRICTION' (with digest of 'e0fb64b1085cc5538970158d05a009c24e276fb94e1a0bf6a528b48fbc4ff526') is enabled with preactivation required
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: info  2019-04-29T20:42:29.075 nodeos    chain_plugin.cpp:396          operator()           ] Support for builtin protocol feature 'GET_SENDER' (with digest of 'f0af56d2c5a48d60a4a5b5c903edfb7db3a736a94ed589d0b797df33ff9d3e1d') is enabled with preactivation required
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: warn  2019-04-29T20:42:29.082 nodeos    chain_plugin.cpp:959          plugin_initialize    ] 13 N5boost16exception_detail10clone_implINS0_19error_info_injectorISt13runtime_errorEEEE: chainbase database format not compatible with this version of chainbase.
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: rethrow chainbase database format not compatible with this version of chainbase.:
Apr 29 20:42:29 jungle-peer6 nodeos[16080]:     {"what":"chainbase database format not compatible with this version of chainbase."}
Apr 29 20:42:29 jungle-peer6 nodeos[16080]:     nodeos  chain_plugin.cpp:959 plugin_initialize
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: error 2019-04-29T20:42:29.082 nodeos    main.cpp:120                  main                 ] 13 N5boost16exception_detail10clone_implINS0_19error_info_injectorISt13runtime_errorEEEE: chainbase database format not compatible with this version of chainbase.
Apr 29 20:42:29 jungle-peer6 nodeos[16080]: rethrow chainbase database format not compatible with this version of chainbase.:
Apr 29 20:42:29 jungle-peer6 nodeos[16080]:     {"what":"chainbase database format not compatible with this version of chainbase."}
Apr 29 20:42:29 jungle-peer6 nodeos[16080]:     nodeos  chain_plugin.cpp:959 plugin_initialize
```

**Comment by matthewdarwin** [comment]
Any plans to fix this issue for 1.8.0-rc3 (or general release?)

**Comment by arhag** [comment]
Should be resolved by #7506 (also #7507).
